README.md

```diff
@@ -8,15 +8,175 @@
+
 
 ## Setup guide
 
-[//]: # (TODO: Add detailed steps to obtain credentials and configure the module.)
+To use the OpenAI Connector, you must have access to the OpenAI API through a [OpenAI Platform account](https://platform.openai.com) and a project under it. If you do not have a OpenAI Platform account, you can sign up for one [here](https://platform.openai.com/signup).
+
+
+1. Open the [OpenAI Platform Dashboard](https://platform.openai.com).
+
+
+2. Navigate to Dashboard -> API keys
+<img src=https://github.com/user-attachments/assets/b2e09c6d-c15f-4cfa-a596-6328b1383162 alt="OpenAI Platform" style="width: 70%;">
+
+
+3. Click on the "Create new secret key" button
+<img src=https://github.com/user-attachments/assets/bf1adab4-5e3f-4094-9a56-5b4c3cc3c19e alt="OpenAI Platform" style="width: 70%;">
+
+
+4. Fill the details and click on Create secret key
+<img src=https://github.com/user-attachments/assets/1c565923-e968-4d5f-9864-7ed2022b8079 alt="OpenAI Platform" style="width: 70%;">
+
+
+5. Store the API key securely to use in your application 
+<img src=https://github.com/user-attachments/assets/bbbf8f38-d551-40ee-9664-f4cf2bd98997 alt="OpenAI Platform" style="width: 70%;">
+
 
 ## Quickstart
 
-[//]: # (TODO: Add a quickstart guide to demonstrate a basic functionality of the module, including sample code snippets.)
+To use the `OpenAI Assistants` connector in your Ballerina application, update the `.bal` file as follows:
+### Step 1: Import the module
+
+Import the `openai.assistants` module.
+
+```ballerina
+import ballerinax/openai.assistants;
+```
+
+### Step 2: Instantiate a new connector
+
+Create a `assistants:ConnectionConfig` with the obtained access token and initialize the connector with it.
+
+```ballerina
+configurable string token = ?;
+
+final assistants:Client openAIAssistant = check new ({
+    auth: {
+        token
+    }
+});
+```
+
+#### Setting HTTP Headers in Ballerina
+
+Calls to the Assistants API require that you pass a beta HTTP header. In Ballerina, you can define the header as follows:
+
+```ballerina
+final map<string|string[]> headers = {
+    "OpenAI-Beta": ["assistants=v2"]
+};
+```
+
+### Step 3: Invoke the connector operations
+
+Now, utilize the available connector operations.
+
+#### Create an assistant
```

Comment:
@SanduniU Usually we add only one simple and most-used use case of the connector in the quickstart. Do we have any particular reason to keep all these multiple samples in here? If not, probably we can remove the others and keep only a basic sample (e.g. creating an assistant)

Reply:
The main use case of assistant is to create a run and get a response from the assistant. In order to create a run, the assistant and the thread must be created. That's why I have added those samples in the quickstart.